src/test/java/cmput402/tdd/AppTest.java

```diff
@@ -227,4 +228,257 @@ public void testDisplayPeople() {
+        }
+        catch(Exception e){
+            assertEquals("Invalid command", e.getMessage());
+        }
+        assertEquals("super apple pie", recipe.getName());
+
+        //Test 13 Invalid command input
+        System.out.println("@@@@@@@@@@@@@@@@@@@@@@@@@");
+        in = new ByteArrayInputStream("foo 2\nsugar\n2.0".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Invalid command", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(1, recipe.items.get(item2).intValue());
+
+        //Test 14 Invalid remove command input
+        in = new ByteArrayInputStream("delete apple\nsugar\n2.0".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Invalid command", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(1, recipe.items.get(item2).intValue());
+
+        //Test 15 Invalid add command input
+        in = new ByteArrayInputStream("add apple\nsugar\n2.0".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Please enter a positive whole number for quantity", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(1, recipe.items.get(item2).intValue());
+
+        //Test 16 Enter invalid rename command with < 40 char
+        in = new ByteArrayInputStream("relabel 'appleappleappleappleappleappleappleapple1'".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Invalid command", e.getMessage());
+        }
+        assertEquals("super apple pie", recipe.getName());
+
+        //Test 17 Rename recipe to string > 40 char with invalid command
+        in = new ByteArrayInputStream("relabel 'appleappleappleappleappleappleappleapple1'".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Invalid command", e.getMessage());
+        }
+        assertEquals("super apple pie", recipe.getName());
+
+        //Test 18 Create invalid item to add
+        in = new ByteArrayInputStream("add 2\n\n2.5".getBytes());
+        scanner = new Scanner(in);
+        try {
+            app.editRecipe(scanner, recipe);
+        }
+        catch(Exception e){
+            assertEquals("Failed to create item for recipe", e.getMessage());
+        }
+        assertEquals(1, recipe.items.size());
+        assertEquals(1, recipe.items.get(item2).intValue());
+
+        //Test 19 Create invalid item to add
```

Comment:
The description of this test should be different than the previous one

Reply:
Fixed in latest commit.